SDWebImage/SDWebImageTransition.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of the SDWebImage package.
+ * (c) Olivier Poitrey <rs@dailymotion.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+#import "SDWebImageCompat.h"
+
+#if SD_UIKIT || SD_MAC
+#import "SDImageCache.h"
+
+// for UIKit(iOS & tvOS), we use `+[UIView transitionWithView:duration:options:animations:completion]` for transition animations.
+// for AppKit(macOS), we use `+[NSAnimationContext runAnimationGroup:completionHandler:]` for transition animations. You can call `+[NSAnimationContext currentContext]` to grab the context during animations block.
+// These transition are provided for basic usage. If you need complicated animation or prefer to directly use Core Animation, consider use `SDWebImageAvoidAutoSetImage` and implement your own after image load finished.
+
+#if SD_UIKIT
+typedef UIViewAnimationOptions SDWebImageAnimationOptions;
+#else
+typedef NS_OPTIONS(NSUInteger, SDWebImageAnimationOptions) {
+    SDWebImageAnimationOptionAllowsImplicitAnimation = 1 << 0, // specify `allowsImplicitAnimation` for the `NSAnimationContext`
+    SDWebImageAnimationOptionNoViewAnimator = 1 << 0 // we use `NSView.animator` instead of view instance itself in the animations block for convenience by default, you can disable this behavior (attention that set image process will not use animator if you disable it)
+};
+#endif
+
+typedef void (^SDWebImageTransitionPreparesBlock)(__kindof UIView * _Nonnull view, UIImage * _Nullable image, NSData * _Nullable imageData, SDImageCacheType cacheType, NSURL * _Nullable imageURL);
+typedef void (^SDWebImageTransitionAnimationsBlock)(__kindof UIView * _Nonnull view, UIImage * _Nullable image);
+typedef void (^SDWebImageTransitionCompletionBlock)(BOOL finished);
+
+@interface SDWebImageTransition : NSObject
+
+/**
+ By default, we set the image to the view at the beginning of the animtions. You can disable this and provide custom set image process
+ */
+@property (nonatomic, assign) BOOL avoidAutoSetImage;
+/**
+ The duration of the transition animation, measured in seconds.
+ */
+@property (nonatomic, assign) NSTimeInterval duration;
+/**
+ The timing function used for all animations within this transition animation (macOS).
+ */
+@property (nonatomic, strong, nullable) CAMediaTimingFunction *timingFunction NS_AVAILABLE_MAC(10_7);
```

Comment:
The minimum macOS version requires is 10.8 currently.

Reply:
This just copied from `NSAnimationContext` and I want to make this not availble on iOS & tvOS & watchOS. If you use `#if SD_MAC`, the generated comments on Swift will leave here. No so good :)